src/span.h

```diff
@@ -27,6 +28,8 @@ class Span
     constexpr C* data() const noexcept { return m_data; }
     constexpr C* begin() const noexcept { return m_data; }
     constexpr C* end() const noexcept { return m_data + m_size; }
+    constexpr C& front() const noexcept { return m_data[0]; }
```

Comment:
Note for reviewers: these functions `front()` and `back()` are no longer used in this PR, but are used in the taproot implementation, so there's no harm in including them here.